ajd/MyDate/lib/MyDate.pm

```diff
@@ -0,0 +1,188 @@
+package MyDate;
+
+use 5.006;
+use strict;
+use warnings;
+
+use Carp qw( croak carp );
+
+=head1 NAME
+
+MyDate - Module with magic date attributes
+
+=head1 VERSION
+
+Version 0.01
+
+=cut
+
+our $VERSION = '0.01';
+
+
+=head1 SYNOPSIS
+
+This module should be able to produce date components without
+any modules being specifically written to do so.
+
+    use MyDate;
+
+    my $date = MyDate->new();
+    print $date->year();
+
+=head1 SUBROUTINES/METHODS
+
+Including these as perldoc does mean there's a bit more manual effort
+involved in updating this package, but tough.
+
+=head2 new
+
+Constructs the date object, set to the current date and time.
+
+=cut
+
+sub new {
+
+  my ( $class ) = @_;
+
+  my @time = localtime;
+
+  # Defining the elements here
+  my $data = {
+    date => $time[3],
+    month => $time[4] + 1,
+    year => $time[5] + 1900,
+  };
+
+  return bless ( $data, $class );
+
+}
+
+=head2 date
+
+Returns the date
+
+=head2 month
+
+Returns the month
+
+=head2 year
+
+Returns the year
+
+=cut
+
+sub AUTOLOAD {
+
+  our $AUTOLOAD;
+  if ( $AUTOLOAD =~ /::(\w+)$/ && $_[0]->{$1} ) {
+    my $field = $1;
+    {
+      no strict 'refs';
+      *$AUTOLOAD = sub {
+        if ( $_[1] ) {
+          $_[0]->{$field} = $_[1];
+        }
+
+        return $_[0]->{$field};
```

Comment:
Yes, but it didn't seem to work when I assigned a $self and had run out of time to figure out why. It didn't make it into the ``no strict 'refs'`` block.